paimon-format/src/test/java/org/apache/paimon/format/parquet/filter/ParquetFilterConverterTest.java

```diff
@@ -0,0 +1,228 @@
+    public void testApplyPredicate(Tuple3 tuple3) {
+        PredicateBuilder builder =
+                new PredicateBuilder(
+                        new RowType(
+                                Collections.singletonList(
+                                        new DataField(0, "fieldName", tuple3.dataType))));
+        test(
+                builder.isNull(0),
+                new ParquetFilters.IsNull("fieldName", tuple3.dataType),
+                tuple3.canPushDown);
+        test(
+                builder.isNotNull(0),
+                new ParquetFilters.Not(new ParquetFilters.IsNull("fieldName", tuple3.dataType)),
+                tuple3.canPushDown);
+        test(
+                builder.equal(0, tuple3.value),
+                new ParquetFilters.Equals("fieldName", tuple3.dataType, tuple3.value),
+                tuple3.canPushDown);
+
+        test(
+                builder.notEqual(0, tuple3.value),
+                new ParquetFilters.Not(
+                        new ParquetFilters.Equals("fieldName", tuple3.dataType, tuple3.value)),
+                tuple3.canPushDown);
+        test(
+                builder.lessThan(0, tuple3.value),
+                new ParquetFilters.LessThan("fieldName", tuple3.dataType, tuple3.value),
+                tuple3.canPushDown);
+        test(
+                builder.lessOrEqual(0, tuple3.value),
+                new ParquetFilters.LessThanEquals("fieldName", tuple3.dataType, tuple3.value),
+                tuple3.canPushDown);
+        test(
+                builder.greaterThan(0, tuple3.value),
+                new ParquetFilters.Not(
+                        new ParquetFilters.LessThanEquals(
+                                "fieldName", tuple3.dataType, tuple3.value)),
+                tuple3.canPushDown);
+        test(
+                builder.greaterOrEqual(0, tuple3.value),
+                new ParquetFilters.Not(
+                        new ParquetFilters.LessThan("fieldName", tuple3.dataType, tuple3.value)),
+                tuple3.canPushDown);
+    }
+
+    static Stream<Tuple3> dataTypeProvider() {
+        return Stream.of(
+                Tuple3.of(new RowType(Lists.newArrayList()), null, false),
+                Tuple3.of(new MultisetType(new TimeType()), null, false),
+                Tuple3.of(new ArrayType(new TimeType()), null, false),
+                Tuple3.of(new MapType(new BooleanType(), new BooleanType()), null, false),
+                Tuple3.of(new BinaryType(), Binary.fromString("a"), true),
+                Tuple3.of(new VarBinaryType(), Binary.fromString("b"), true),
+                Tuple3.of(new CharType(), Binary.fromString("c"), true),
+                Tuple3.of(new VarCharType(), Binary.fromString("paimon"), true),
+                Tuple3.of(new TimeType(), LocalTime.now(), true),
+                Tuple3.of(new DateType(), LocalDate.now(), true),
+                Tuple3.of(new TimestampType(), LocalDateTime.now(), true),
```

Comment:
No test for `TimestampType(9)`

Reply:
I add it.
and I test that, although `TimestampType(9)` is converted to `Binary` , It can still be pushdown for `lessThan`,`equal`,`greaterThan`, and get the correct result.

Reply:
I don't think so, you can take a look to `ParquetRowDataWriter.timestampToInt96`, that is not Binary order.